manta-js/package/sdk.ts

```diff
@@ -0,0 +1,626 @@
+    async privateTransferSend(assetId: BN, amount: BN, toPrivateAddress: Address, polkadotSigner:Signer, polkadotAddress:Address): Promise<void> {
+        const signed = await this.privateTransferBuild(assetId,amount,toPrivateAddress,polkadotSigner,polkadotAddress);
+        await sendTransaction(polkadotAddress,signed);
+    }
+
+    /// Builds a "Private Transfer" transaction for any fungible token.
+    /// Note: This transaction is not published to the ledger.
+    async privateTransferBuild(assetId: BN, amount: BN, toPrivateAddress: Address, polkadotSigner:Signer, polkadotAddress:Address): Promise<SignedTransaction> {
+        if (!this.walletIsBusy) {
+            this.walletIsBusy = true;
+            await this._setPolkadotSigner(polkadotSigner, polkadotAddress);
+            const transaction = await privateFungibleTransfer(this.api, this.wasm, assetId, amount, toPrivateAddress);
+            const signResult = await signTransaction(this.api, this.wasm, this.wasmWallet, transaction.assetMetadataJson, transaction.transaction, this.network);
+            this.walletIsBusy = false;
+            return signResult;
+        } else {
+            console.error("Wallet is currently busy!");
+        }
+    }
+
+    /// Executes a "To Public" transaction for any fungible token.
+    async toPublicSend(assetId: BN, amount: BN, polkadotSigner:Signer, polkadotAddress:Address): Promise<void> {
+        const signed = await this.toPublicBuild(assetId,amount,polkadotSigner, polkadotAddress);
+        await sendTransaction(polkadotAddress,signed);
+    }
+
+    /// Builds and signs a "To Public" transaction for any fungible token.
+    /// Note: This transaction is not published to the ledger.
+    async toPublicBuild(assetId: BN, amount: BN, polkadotSigner:Signer, polkadotAddress:Address): Promise<SignedTransaction> {
+        if (!this.walletIsBusy) {
+            this.walletIsBusy = true;
+            await this._setPolkadotSigner(polkadotSigner, polkadotAddress);
+            const transaction = await toPublicFungible(this.api, this.wasm, assetId, amount);
+            const signResult = await signTransaction(this.api, this.wasm, this.wasmWallet, transaction.assetMetadataJson, transaction.transaction, this.network);
+            this.walletIsBusy = false;
+            return signResult;
+        } else {
+            console.error("Wallet is currently busy!");
+        }
+    }
+
+    /// Executes a public transfer.
+    async publicTransfer(assetId: BN, amount: BN, destinationAddress: Address, senderAddress:Address, polkadotSigner:Signer): Promise<void> {
+        await this._setPolkadotSigner(polkadotSigner, senderAddress);
+        await publicFungibleTransfer(this.api, senderAddress, assetId, destinationAddress, amount);
+    }
+}
+
+/// Returns the corresponding blockchain connection URL from Environment
+/// and Network values. 
+function envUrl(env: Environment, network: Network): string {
```

Comment:
I think all of the functions not in the class that are called by the class actually belong inside the class. Everything else is probably a utility and belongs in a separate file. There is probably some better way to organize / decompose all the functionality in this file, but I think the current division of stuff in vs not in the class is a bit arbitrary / confusing. So I think moving stuff into the class is a decent next step.

Some like `envUrl` here can be static methods. Others, like `getPrivateAddress` can just be moved into the calling function inside the class (e.g. privateAddress`). In terms of organization, I would put functions we intend to publicly expose at the top of the class. Other functions we can denote as private (using typescript I believe, or otherwise with a leading underscore).